app/Providers/AppServiceProvider.php

```diff
@@ -2,43 +2,43 @@
 
 namespace Adshares\Adserver\Providers;
 
+use Adshares\Adserver\Services\Adpay;
 use Adshares\Adserver\Services\Adselect;
-use Adshares\Esc\Esc;
-
+use Adshares\Ads\AdsClient;
+use Adshares\Ads\Driver\CliDriver;
 use Illuminate\Support\ServiceProvider;
 
 class AppServiceProvider extends ServiceProvider
 {
     /**
      * Bootstrap any application services.
-     *
-     * @return void
      */
     public function boot()
     {
-        //
     }
 
     /**
      * Register any application services.
-     *
-     * @return void
      */
     public function register()
     {
+        $this->app->bind(Adpay::class, function ($app) {
+            return new Adpay(config('app.adpay_endpoint'), config('app.debug'));
+        });
         $this->app->bind(Adselect::class, function ($app) {
             return new Adselect(config('app.adselect_endpoint'), config('app.debug'));
         });
-        $this->app->bind(Esc::class, function ($app) {
-            // Esc($walletCommand, $workingDir, $address, $secret, $host, $port)
-            return new Esc(
-                config('app.adshares_wallet'),
-                config('app.adshares_workdir'),
+        $this->app->bind(AdsClient::class, function ($app) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "$app". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1172)